libs/langchain-azure-cosmosdb/src/azure_cosmosdb_mongodb.ts

```diff
@@ -0,0 +1,496 @@
+  readonly connectionString?: string;
+  readonly databaseName?: string;
+  readonly collectionName?: string;
+  readonly indexName?: string;
+  readonly textKey?: string;
+  readonly embeddingKey?: string;
+  readonly indexOptions?: AzureCosmosDBMongoDBIndexOptions;
+}
+
+/**
+ * Azure Cosmos DB for MongoDB vCore vector store.
+ * To use this, you should have both:
+ * - the `mongodb` NPM package installed
+ * - a connection string associated with a MongoDB VCore Cluster
+ *
+ * You do not need to create a database or collection, it will be created
+ * automatically.
+ *
+ * You also need an index on the collection, which is by default be created
+ * automatically using the `createIndex` method.
+ */
+export class AzureCosmosDBMongoDBVectorStore extends VectorStore {
+  get lc_secrets(): { [key: string]: string } {
+    return {
+      connectionString: "AZURE_COSMOSDB_MONGODB_CONNECTION_STRING",
+    };
+  }
+
+  private connectPromise: Promise<void>;
+
+  private initPromise: Promise<void>;
+
+  private readonly client: MongoClient | undefined;
+
+  private database: Db;
+
+  private collection: Collection<MongoDBDocument>;
+
+  readonly indexName: string;
+
+  readonly textKey: string;
+
+  readonly embeddingKey: string;
+
+  private readonly indexOptions: AzureCosmosDBMongoDBIndexOptions;
+
+  /**
+   * Initializes the AzureCosmosDBMongoDBVectorStore.
+   * Connect the client to the database and create the container, creating them if needed.
+   * @returns A promise that resolves when the AzureCosmosDBMongoDBVectorStore has been initialized.
+   */
+  initialize: () => Promise<void>;
+
+  _vectorstoreType(): string {
+    return "azure_cosmosdb_mongodb";
+  }
+
+  constructor(
+    embeddings: EmbeddingsInterface,
+    dbConfig: AzureCosmosDBMongoDBConfig
+  ) {
+    super(embeddings, dbConfig);
+
+    const connectionString =
+      dbConfig.connectionString ??
+      getEnvironmentVariable("AZURE_COSMOSDB_MONGODB_CONNECTION_STRING");
+
+    if (!dbConfig.client && !connectionString) {
+      throw new Error(
+        "AzureCosmosDBMongoDBVectorStore client or connection string must be set."
+      );
+    }
+
+    if (!dbConfig.client) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      this.client = new MongoClient(connectionString!, {
+        appName: "langchainjs",
+      });
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const client = dbConfig.client || this.client!;
+    const databaseName = dbConfig.databaseName ?? "documentsDB";
+    const collectionName = dbConfig.collectionName ?? "documents";
+    this.indexName = dbConfig.indexName ?? "vectorSearchIndex";
+    this.textKey = dbConfig.textKey ?? "textContent";
+    this.embeddingKey = dbConfig.embeddingKey ?? "vectorContent";
+    this.indexOptions = dbConfig.indexOptions ?? {};
+
+    // Deferring initialization to the first call to `initialize`
+    this.initialize = () => {
```

Comment:
Would prefer this as a static method like this:

https://v02.api.js.langchain.com/classes/langchain_community_vectorstores_pgvector.PGVectorStore.html#initialize

Since I'm not sure this will show up well in API refs, but won't block on it

Reply:
Would you be ok with a middle ground approach?
- make the current `initialize` method I made internal
- add a new static `initialize` method the calls constructor + init/connect the db connection
- keep the "automatic" init when calling any VS method in case it wasn't done before (for compatibility / simplicity)
- explain how the init works in the docs

My reasoning: I'd like to make the migration from the previous implementation painless (ie keep the auto init) and also make it work for folks who skim quickly through docs and just call the constructor without using the `initialize` method